File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/common/client/jaxrs/tls/TLSAuth.java
package co.uk.swarmbit.docker.api.common.client.jaxrs.tls;

import co.uk.swarmbit.docker.config.DockerClientConfig;
import co.uk.swarmbit.exception.UnsupportedConfiguration;
import org.apache.commons.lang3.StringUtils;

public enum TLSAuth {
    TLS,
    TLS_AUTHENTICATE_SERVER,
    TLS_AUTHENTICATE_CLIENT,
    TLS_AUTHENTICATE_SERVER_CLIENT;

    public static TLSAuth getTLSMode(DockerClientConfig config){
        if (StringUtils.equals(config.getTlsMode(), "tls")) {
            if (config.getTlsCert().isPresent() && config.getTlsKey().isPresent()) {
                return TLS_AUTHENTICATE_CLIENT;
            }
            return TLS;
        } else if (StringUtils.equals(config.getTlsMode(), "tlsverify")) {
            if (config.getTlsCacert().isPresent()) {
                if (config.getTlsCert().isPresent() && config.getTlsKey().isPresent()) {
                    return TLS_AUTHENTICATE_SERVER_CLIENT;
                }
                return TLS_AUTHENTICATE_SERVER;
            }
            throw new UnsupportedConfiguration("Should provide tlscacert when using tlsverify");
        }
        throw new UnsupportedConfiguration("TLS mode should be tls or tlsverify");
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/client/jaxrs/TLSAuth.kt
package com.swarmbit.docker.api.client.jaxrs

import com.swarmbit.docker.api.config.DockerClientConfig
import com.swarmbit.docker.api.exception.UnsupportedConfiguration

enum class TLSAuth {
    TLS, TLS_AUTHENTICATE_SERVER, TLS_AUTHENTICATE_CLIENT, TLS_AUTHENTICATE_SERVER_CLIENT;

    companion object {
        fun getTLSMode(config: DockerClientConfig): TLSAuth {
            if (config.tlsMode == "tls") {
                return if (config.tlsCacert != null && config.tlsKey != null) {
                    TLS_AUTHENTICATE_CLIENT
                } else TLS
            } else if (config.tlsMode == "tlsverify") {
                if (config.tlsCacert != null) {
                    return if (config.tlsCert != null && config.tlsKey != null) {
                        TLS_AUTHENTICATE_SERVER_CLIENT
                    } else TLS_AUTHENTICATE_SERVER
                }
                throw UnsupportedConfiguration("Should provide tlscacert when using tlsverify")
            }
            throw UnsupportedConfiguration("TLS mode should be tls or tlsverify")
        }
    }
}
